lib/matplotlib/axes/_axes.py

```diff
@@ -2440,8 +2449,12 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
         elif orientation == 'horizontal':
             datavalues = width
 
-        bar_container = BarContainer(patches, errorbar, datavalues=datavalues,
-                                     orientation=orientation, label=label)
+        if data_empty:
+            bar_container = BarContainer([], [], datavalues=[],
```

Comment:
```suggestion
            # we've added a dummy Rectangle to the Axes above in case of
            # empty data to keep the legend correct. However, we do not want
            # to return it here. There are no values and thus there should be
            # no patches in the BarContainer.
            bar_container = BarContainer([], [], datavalues=[],
```